File: 1. Two Sum/Solution.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] res = new int[2];
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < numbers.length; i++) {
            if (!map.containsKey(numbers[i])) {
                map.put(target - numbers[i], i+1);
            } else {
                res[0] = map.get(numbers[i]);
                res[1] = i+1;
                return res;
            }
        }
        return res;
    }
}

// Add a comment.

File: 1. Two Sum/Solution.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer[]}
    def twoSum(self, nums, target):
        mydic = {}
        res = [-1] * 2
        for i in xrange(len(nums)):
            if nums[i] not in mydic:
                mydic[target - nums[i]] = i+1
            else:
                res[0] = mydic[nums[i]]
                res[1] = i+1
                break
        return res